swap_meet/vendor.py

```diff
@@ -0,0 +1,102 @@
+class Vendor:
+    def __init__(self, inventory=None):
+        if inventory == None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+    
+    """
+    This method adds the provided argument to that object's inventory,
+    and returns the argument. 
+    """
+    def add(self,item):
+        self.inventory.append(item)
+        return item
+
+    """
+    This method removes the provided argument, if it is already
+    in the inventory, from that object's inventory and returns the argument.
+    Otherwise, this method returns False.
+    """
+    def remove(self,item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+
+    """
+    This function iterates through all the items in the specified
+    object's inventory and appends items whose category
+    matches the provided category argument and returns the
+    newly modified list.
+    """
+    def get_by_category(self,category):
+        category_specific_list = []
+        for item in self.inventory:
+            if item.category == category:
+                category_specific_list.append(item)
+        return category_specific_list
+
+    """
+    This function swaps selected items (from the current object's inventory
+    and anther instance of Vendor's inventory), but only if the
+    items are actually in their respective inventories. 
+    If an item is missing, the method returns False.
+    """
+    def swap_items(self, Vendor, my_item, their_item):
+        if my_item in self.inventory and their_item in Vendor.inventory:
+            self.inventory.remove(my_item)
+            self.inventory.append(their_item)
+            Vendor.inventory.remove(their_item)
+            Vendor.inventory.append(my_item)
+            return True
+        else:
+            return False
```

Comment:
I'd advise against changing class attributes as `self`  represents an instance of the `Vendor` class so when you remove an item from `Vendor`, you're also removing it from all **instances** of Vendor.